cyclonedx-lib/sign_src/TemurinSignSBOM.java

```diff
@@ -0,0 +1,161 @@
+    private static final Logger LOGGER = Logger.getLogger(TemurinSignSBOM.class.getName());
+
+    private TemurinSignSBOM() {
+    }
+
+    /**
+     * Main entry.
+     * @param args Arguments for sbom operation.
+     */
+    public static void main(final String[] args) throws Exception {
+        String cmd = null;
+        String privateKeyFile = null;
+        String publicKeyFile = null;
+        String fileName = null;
+
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equals("--jsonFile")) {
+                fileName = args[++i];
+            } else if (args[i].equals("--privateKeyFile")) {
+                privateKeyFile = args[++i];
+            } else if (args[i].equals("--publicKeyFile")) {
+                publicKeyFile = args[++i];
+            } else if (args[i].equals("--signSBOM")) {
+                cmd = "signSBOM";
+            } else if (args[i].equals("--verifySignature")) {
+                cmd = "verifySignature";
+            } else if (args[i].equals("--verbose")) {
+                verbose = true;
+            }
+        }
+
+
+        if (cmd.equals("signSBOM")) {
+            Bom bom = signSBOM(fileName, privateKeyFile);
+            writeJSONfile(bom, fileName);
+        } else if (cmd.equals("verifySignature")) {
+            boolean isValid = verifySignature(fileName, publicKeyFile);
+            System.out.println("Signature verification result: " + (isValid ? "Valid" : "Invalid"));
+            if (!isValid) {
+                System.exit(1);
+            }
+        }
+    }
+
+    static Bom signSBOM(final String jsonFile, final String pemFile) throws IOException, GeneralSecurityException, ParseException {
+        // Read the JSON file to be signed
+        Bom bom = readJSONfile(jsonFile);
+        String sbomDataToSign = generateBomJson(bom);
+
+        // Read the private key
+        KeyPair signingKey = PEMDecoder.getKeyPair(Files.readAllBytes(Paths.get(pemFile)));
+
+        // Sign the JSON data
+        String signedData = new JSONObjectWriter(JSONParser.parse(sbomDataToSign))
+                .setSignature(new JSONAsymKeySigner(signingKey.getPrivate()))
+                .serializeToString(JSONOutputFormats.PRETTY_PRINT);
+
+        JsonParser parser = new JsonParser();
+        Bom signedBom = parser.parse(new StringReader(signedData));
+        return signedBom;
+    }
+
+    static String generateBomJson(final Bom bom) {
+        BomJsonGenerator bomGen = BomGeneratorFactory.createJson(CycloneDxSchema.Version.VERSION_14, bom);
+        String json = bomGen.toJsonString();
+        return json;
+    }
+
+    static void writeJSONfile(final Bom bom, final String fileName) {
+        // Creates testJson.json file
+        String json = generateBomJson(bom);
+        try (FileWriter file = new FileWriter(fileName)) {
+            file.write(json);
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error writing JSON file " + fileName, e);
```

Comment:
This is swallowing the exception, it should either re-throw the exception, or system exit, otherwise it could write a invalid file but the cmd still succeed.
Either:
1. Make it return a boolean Succeed flag, and caller checks and does System.exit(1)
2. Make it do a System.exit(1)
3. Re-throw the exception

My preference is probably (1)